### PR TITLE
added make_radar_entities as point_cloud2_utils

### PR DESCRIPTION
**Pull request by ClementBotty**
Hi Iurii, I added 2 functions essentials for sending entities through DDS, I want to merge it to master as it seems to be the most up-to-date branch. (Feel free to change it to develop). 

The two function has been tested and work perfectly fine. 

I don't know to do it more concisely when defining the offset for each field. 

:warning: It means that provizio_dds will have to be recompiled before Thursday's tests.

### CODE REVIEW

File: python/point_cloud2.py [hunk 1]
```diff
@@ -357,6 +357,120 @@ def make_radar_point_cloud(header: Header, points: Iterable, is_dense: bool = Tr
     return create_cloud(header, fields, points, is_dense)
 
 
+def create_entities_cloud(
```
Comment (iurii-provizio) [inline python/point_cloud2.py]: @ClementBotty I don't get why this function is even needed. Just a single call. Besides, `:param is_dense: True if there are no invalid points` us described but not present

File: python/point_cloud2.py [hunk 2]
```diff
@@ -357,6 +357,120 @@ def make_radar_point_cloud(header: Header, points: Iterable, is_dense: bool = Tr
...
+                   the fields parameter)
+    :param is_dense: True if there are no invalid points
+    :return: The point cloud as provizio_dds.PointCloud2
+    """
+    entities_cloud = create_cloud(header, fields, entities)
+
+    return entities_cloud
+
+
+def make_radar_entities(
```
Comment (iurii-provizio) [inline python/point_cloud2.py:360]: @ClementBotty Please indeed also add an additional test to `test/python/pointcloud2_test.py`

File: python/point_cloud2.py [hunk 3]
```diff
@@ -357,6 +357,120 @@ def make_radar_point_cloud(header: Header, points: Iterable, is_dense: bool = Tr
...
+    # orientation (x, y, z, w)
+    fields[7] = PointField()
+    fields[7].offset(25)
+    fields[7].count(4)
+    fields[7].datatype(FLOAT32)
+    fields[7].name("orientation")
+
+    # size (x, y, z)
+    fields[8] = PointField()
+    fields[8].offset(41)
```
Comment (iurii-provizio) [inline python/point_cloud2.py]: @ClementBotty Not great specifying offsets directly. I suggest adding a counter starting at 0, then increment it after every field specification. That way reordering fields will much easier, besides it's much simpler to check the logic and correctness of offsets.

File: python/point_cloud2.py (hunk 1 above)
Comment (ClementBotty) [reply python/point_cloud2.py]: I removed `create entities_cloud` and now I don't use 'is_dense' anymore.

File: python/point_cloud2.py (hunk 2 above)
Comment (ClementBotty) [reply python/point_cloud2.py:360]: I added tests to the repository, to check metadata and data for embedded entities in PointCloud2.

File: python/point_cloud2.py (hunk 3 above)
Comment (ClementBotty) [reply python/point_cloud2.py]: I added a variable that count the offset, and is explicit. 
For example after defining size, which contains 3 elements of FLOAT32, I did `offset += 3 * 4`

File: python/point_cloud2.py [hunk 4]
```diff
@@ -357,6 +357,111 @@ def make_radar_point_cloud(header: Header, points: Iterable, is_dense: bool = Tr
...
+
+    # radar_relative_radial_velocity
+    fields[5] = PointField()
+    fields[5].offset(offset)
+    fields[5].count(1)
+    fields[5].datatype(FLOAT32)
+    fields[5].name("radar_relative_radial_velocity")
+    offset += 4
+
+    # radar_relative_radial_velocity
```
Comment (iurii-provizio) [inline python/point_cloud2.py]: @ClementBotty Icorrect comment

File: test/python/pointcloud2_test.py [hunk 5]
```diff
@@ -53,4 +53,34 @@
...
+
+print("Creating a PointCloud2 embedding entities... ")
+entities = [[99, 4, 20.5, -2.0, 1.0, 10.2, 25.0, 0, 0, 0, 1, 2, 5, 2, 254, 254],
+            [100, 2, 10.0, 2.0, 1.0, -10.0, 0.0, 0, 0, 0, 1, float('nan'), float('nan'), float('nan'), 254, 12]]
+
+entities_cloud = provizio_dds.point_cloud2.make_radar_entities(provizio_dds.point_cloud2.make_header(10, 20, "test_entities"), entities)
+
+# Check the entities cloud
+print("Checking the embedded entities in PointCloud2 metadata...")
+assert entities_cloud.fields()[0].name() == "track_id"
```
Comment (iurii-provizio) [inline test/python/pointcloud2_test.py]: `entity_id` I guess?

Comment (ClementBotty) [reply test/python/pointcloud2_test.py]: Good catch Fixed :+1:

File: python/point_cloud2.py (hunk 4 above)
Comment (ClementBotty) [reply python/point_cloud2.py]: Good catch Fixed 👍

File: test/python/pointcloud2_test.py [hunk 6]
```diff
@@ -53,4 +53,33 @@
...
+assert entities_cloud.fields()[6].name() == "ground_relative_radial_velocity"
+assert entities_cloud.fields()[7].name() == "orientation"
+assert entities_cloud.fields()[8].name() == "size"
+assert entities_cloud.fields()[9].name() == "entity_confidence"
+assert entities_cloud.fields()[10].name() == "entity_class_confidence"
+
+# Read it
+print("Reading the PointCloud2 entities and checking its data...")
+
+read_entities = provizio_dds.point_cloud2.read_points(entities_cloud)
```
Comment (iurii-provizio) [inline test/python/pointcloud2_test.py:82]: @ClementBotty It'll likely work, but it doesn't fix the issue you faced in the previous version of that tests. I'm OK not to do it in this PR, but please make a Jira bug.

**Review by iurii-provizio — APPROVED** [review]

**Pull request merged**